### PR TITLE
adjust continuation frame header length

### PR DESCRIPTION
**Pull request by CLFutureX**
Motivation:

Padding was removed from CONTINUATION frame in http2-spec, 
at the same time , the continuation frame header should be adjusted,
which can also to reduce the allocation of bytes.

### CODE REVIEW

File: codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java [hunk 1]
```diff
@@ -583,7 +583,7 @@ private ChannelFuture writeContinuationFrames(ChannelHandlerContext ctx, int str
                     // The frame header is different for the last frame, so re-allocate and release the old buffer
                     flags = flags.endOfHeaders(true);
                     buf.release();
-                    buf = ctx.alloc().buffer(CONTINUATION_FRAME_HEADER_LENGTH);
+                    buf = ctx.alloc().buffer(FRAME_HEADER_LENGTH);
```
Comment (normanmaurer) [inline codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java:586]: Shouldn't we better adjust `CONTINUATION_FRAME_HEADER_LENGTH` to be `==` `FRAME_HEADER_LENGTH`

**Review by normanmaurer — CHANGES_REQUESTED** [review]

File: codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java (hunk 1 above)
Comment (CLFutureX) [reply codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java]: > Shouldn't we better adjust `CONTINUATION_FRAME_HEADER_LENGTH` to be `==` `FRAME_HEADER_LENGTH`

Yes, the first modification was to CONTINUATION_FRAME_HEADER_LENGTH. The check failed at that time, and there was concern about affecting other callers, but upon careful consideration, adjust CONTINUATION_FRAME_HEADER_LENGTH would be better

Comment (CLFutureX) [reply codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java]: > > Shouldn't we better adjust `CONTINUATION_FRAME_HEADER_LENGTH` to be `==` `FRAME_HEADER_LENGTH`
> 
> Yes, the first modification was to CONTINUATION_FRAME_HEADER_LENGTH. The check failed at that time, and there was concern about affecting other callers, but upon careful consideration, adjust CONTINUATION_FRAME_HEADER_LENGTH =  FRAME_HEADER_LENGTH would be better

**Comment by normanmaurer** [comment]
I would argue we should ignore the Revapi failure here as at worse it will just use the old value and so allocate 1 extra byte until things are re-compiled. @chrisvest WDYT ?

**Comment by chrisvest** [comment]
@normanmaurer I agree. Revapi is flagging it because such a change _could_ be problematic, but this case should be fine as you say.
@CLFutureX Just add a difference to be ignored in the top-level `pom.xml` file: https://github.com/netty/netty/blob/4.1/pom.xml#L1297

**Comment by CLFutureX** [comment]
> Padding was removed from CONTINUATION frame in http2-spec,



> @normanmaurer I agree. Revapi is flagging it because such a change _could_ be problematic, but this case should be fine as you say. @CLFutureX Just add a difference to be ignored in the top-level `pom.xml` file: https://github.com/netty/netty/blob/4.1/pom.xml#L1297

okay，i fix it

**Comment by normanmaurer** [comment]
@CLFutureX Thanks a lot!

**Pull request merged**